src/config.py

```diff
@@ -490,36 +524,35 @@ def set_zk_jaas_config(self) -> None:
                 password="{self.zookeeper_config['password']}";
             }};
         """
-        push(
+        safe_push_to_file(
             container=self.container,
             content=jaas_config,
             path=self.zk_jaas_filepath,
         )
 
     def set_server_properties(self) -> None:
         """Writes all Kafka config properties to the `server.properties` path."""
-        push(
+        safe_push_to_file(
+            container=self.container,
             content="\n".join(self.server_properties),
             path=self.server_properties_filepath,
-            container=self.container,
         )
 
     def set_client_properties(self) -> None:
         """Writes all client config properties to the `client.properties` path."""
-        push(
+        safe_push_to_file(
+            container=self.container,
             content="\n".join(self.client_properties),
             path=self.client_properties_filepath,
-            container=self.container,
         )
 
     def set_environment(self) -> None:
         """Writes the env-vars needed for SASL/SCRAM auth to `/etc/environment` on the unit."""
-        opts_string = " ".join(self.extra_args)
-        push(
-            content=f"KAFKA_OPTS={opts_string}\nLOG_DIR={LOGS_PATH}\nJAVA_HOME={JAVA_HOME}",
-            path="/etc/environment",
-            container=self.container,
-        )
+        java_home = f"JAVA_HOME='{JAVA_HOME}'"
+        log_dir = f"LOG_DIR='{LOGS_PATH}'"
+        updated_env_list = [self.kafka_opts, self.jmx_opts, self.log4j_opts, java_home, log_dir]
```

Comment:
Ugh, properties are strings, so it would look a bit weird:
```python
base_env = (
    [
        f"JAVA_HOME='{JAVA_HOME}'",
        f"LOG_DIR='{LOGS_PATH}'",
    ]
    + [self.kafka_opts]
    + [self.jmx_opts]
    + [self.log4j_opts]
)
```